src/util/api.js

```diff
@@ -65,6 +57,13 @@ const post = (path, body) => {
         return res.json();
       }
       return res.text();
+    })
+    .then(res => {
+      if (res.status >= 400) {
```

Comment:
What happens if text() is used? I.e. 
- `res.text().then(data => data.status)` or 
- `res.text().then(deserialize).then(data => data.status)`

I assume that the status is in those cases "undefined". 

I kind of feel that this should be handled differently. Maybe you could follow the "serialize" pattern if the error is only possible with 'application/json' format type?

Reply:
@Gnito good catch! I I'm not sure how this should be handled. What do you mean by following the "serialize" pattern?

Reply:
Sorry, I meant to write "deserialize" to refer to existing function. I.e. I was referring to having a completely separate function that handles error printing (similarly as `res.text().then(deserialize)`). 

Anyway, there could be a custom path for error handling when `if (res.status >= 400) {` the "throw" is anyway breaking the "then" chain.

Reply:
I think you could write something like:

```js
if (res.status >= 400) {
  return res.json().then(throwErrorHelper);
} 
```

Reply:
@Gnito That makes sense, thanks for explaining! I changed the error handling so do you think it's ok now?